### ISSUE TITLE
ESLint giving false positives for unused vars in typescript code

### ISSUE THREAD

**Issue by ChrisOtto1014** [microsoft/vscode-eslint#1163]
I am using Visual Studio Code with the ESLint extension and am getting false positives on imports being reported as violating the @typescript-eslint/no-unused-vars rule.

Visual Studio Code version info:

Version: 1.52.1 (user setup)
Commit: ea3859d4ba2f3e577a159bc91e3074c5d85c0523
Date: 2020-12-16T16:34:46.910Z
Electron: 9.3.5
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Windows_NT x64 10.0.19042

ESLint v2.1.14

settings.json and eslintrc files are attached. The same code and eslintrc show no warnings when using IntelliJ Ultimate.

![image](https://user-images.githubusercontent.com/20071696/105392282-05ef2b80-5be9-11eb-871e-883d9c0c1d61.png)

eslintrc:
```
{
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended",
        "prettier/@typescript-eslint",
        "plugin:prettier/recommended",
        "plugin:react/recommended"
    ],
    "env": {
        "browser": true,
        "node": true,
        "es6": true
    },
    "parser": "@typescript-eslint/parser",
    "plugins": ["@typescript-eslint", "react"],
    "parserOptions": {
        "ecmaVersion": 2018, // Allows for the parsing of modern ECMAScript features
        "sourceType": "module", // Allows for the use of imports
        "ecmaFeatures": {
            "jsx": true // Allows for the parsing of JSX
        }
    },
    "settings": {
        "react": {
            "version": "detect" // Tells eslint-plugin-react to automatically detect the version of React to use
        }
    }
}
```
settings.json

```
{
    "editor.minimap.enabled": false,
    "editor.fontSize": 11,
    "[javascript]": {
        "editor.defaultFormatter": "esbenp.prettier-vscode"
    },
    "git.enableSmartCommit": true,
    "files.autoSave": "off",
    "vim.useSystemClipboard": true,
    "eslint.alwaysShowStatus": true,
    "eslint.codeAction.showDocumentation": {
        "enable": true
    },
    "[typescriptreact]": {
        "editor.defaultFormatter": "esbenp.prettier-vscode"
    },
    "editor.formatOnSave": true
}
```

**Comment by dbaeumer** [comment]
@ChrisOtto1014 does the validation happen correctly when you use eslint in the terminal. If this is the case can you please provide me with a GitHub repository I can clone that demos what you are experiencing.

**Comment by lifeiscontent** [comment]
@dbaeumer I have a similar issue, I can give you access to the codebase I'm seeing it in if you have time to debug it

**Comment by dbaeumer** [comment]
@lifeiscontent does validation happen correctly for you in the terminal? If so access to the repository you are seeing it is apprecaited.

**Comment by lifeiscontent** [comment]
nope, its only via your plugin. I'll send you an invite, I'm seeing it in this branch `feature/components` on file `packages/react-components/src/Sliderdiscrete/Sliderdiscrete.tsx` line 89 `React.Children.map`

**Comment by dbaeumer** [comment]
OK. Will wait for the invite :-)

**Comment by lifeiscontent** [comment]
@dbaeumer already sent during the comment I left. https://github.com/lifeiscontent/lifetime/invitations

**Comment by lifeiscontent** [comment]
@dbaeumer any updates here?

**Comment by dbaeumer** [comment]
I see no unused var error. Here is what I see:

<img width="606" alt="capture" src="https://user-images.githubusercontent.com/1931590/110131325-5225b200-7dca-11eb-8466-03dd157efb56.png">

**Comment by lifeiscontent** [comment]
@dbaeumer yep, but if you run the lint in the terminal that doesn't show up as an error. its exclusive to the vscode plugin.

**Comment by lifeiscontent** [comment]
@dbaeumer there are other issues that you've closed in the past that are similar but failed to provide a reproducible example. So this is what I'm providing for you, however I guess its not this exact issue as stated above.

**Comment by dbaeumer** [comment]
Actually I see the same error when running ESLint in the terminal for the given file using:

```
./node_modules/.bin/eslint packages/react-components/src/Sliderdiscrete/Sliderdiscrete.tsx 
```

**Comment by lifeiscontent** [comment]
@dbaeumer is there a possibility that vscode-eslint is not finding the root eslint config?

**Comment by lifeiscontent** [comment]
@dbaeumer I've been chatting with @bradzacher as well on this issue about it: https://github.com/typescript-eslint/typescript-eslint/issues/3053

**Comment by dbaeumer** [comment]
@lifeiscontent VS Code eslint uses the eslint npm module under the hoods. So if the eslint command in a terminal (e.g. `./node_modules/.bin/eslint packages/react-components/src/Sliderdiscrete/Sliderdiscrete.tsx`) finds the config or doesn't find it then the same is true for the extension.

**Comment by dbaeumer** [comment]
This issue has been closed automatically because it needs more information and has not had recent activity. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Happy Coding!

**Comment by YashKumarVerma** [comment]
I am also facing the same issue.

**Comment by karimcambridge** [comment]
Hey I noticed this issue over the past few months.

It happens in terminal and in vscode.. its happening on all my react (native) projects, for e.g look at this screenshot

```js
import '../styles/globals.css';
import { ChakraProvider } from '@chakra-ui/react';

import { useAuth, AuthProvider } from '../contexts/auth';

const MyApp = ({ Component, pageProps }) => {
	return (
		<AuthProvider>
			<ChakraProvider>
				<Component {...pageProps} />
			</ChakraProvider>
		</AuthProvider>
	);
};

export default MyApp;
```

**
![image](https://user-images.githubusercontent.com/4283893/118579267-bc0fef80-b75b-11eb-8e9a-3326a7dcbfc6.png)
**

It seems to happen to object destructuring only.

**Comment by roadroot** [comment]
Hello, I am facing the same issue. Could you please open this issue as it is faced by many persons.

**Comment by roadroot** [comment]
I resolved this issue by adding :
`    "@typescript-eslint/no-unused-vars": "warn",`
